### PR TITLE
Updated consensus cnv wxs wgs

### PR DESCRIPTION
**Pull request by zhangb1**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- updated the scripts from OT and PBTA for WXS and WGS consensus
- updated the CWL tools and workflow to use the new command line

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Environment:
* Test files:

the cwl has been tested in cavatica project :  https://cavatica.sbgenomics.com/u/zhangb1/data-assembly-dev/tasks/f89058d6-08ad-4bc4-9263-ee8e181737d2

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] I have committed any related changes to the PR

### CODE REVIEW

**Comment by migbro** [comment]
Ok, it's been added to our review session for tomorrow

File: cwl/tools/merge_seg.cwl [hunk 1]
```diff
@@ -15,9 +15,10 @@ arguments:
 
       egrep -v "loc.start" $(inputs.input_seg.path) |gzip >temp.gz && cat $(inputs.input_previous_merged_seg.path) temp.gz > pbta-cnv-consensus.$(inputs.biospecimen_id).seg.gz && rm temp.gz
 inputs:
-  input_seg: File
-  input_previous_merged_seg: File
+  input_seg: File?
+  input_previous_merged_seg: File?
   biospecimen_id: string
+  run_WGS_or_WXS: { type: { type: 'enum', name: run_WGS_or_WXS, symbols: ["WGS", "WXS"] } }
```
Comment (migbro) [inline cwl/tools/merge_seg.cwl:21]: Merged seg seems to have been set in the wf to only run when input is WGS. Shouldn't the output also be WXS?

**Review by migbro — COMMENTED** [review]
It seems ok overall, but see my comment. Maybe do a run with WXS input, test that flag, just to make sure it runs runs as expected when input is not WGS.

File: cwl/tools/merge_sv.cwl [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class: CommandLineTool
+cwlVersion: v1.0
+id: merge_sv
+requirements:
+  - class: ShellCommandRequirement
+  - class: InlineJavascriptRequirement
+  - class: DockerRequirement
+    dockerPull: 'zhangb1/data-assembly'
```
Comment (mark-welsh) [inline cwl/tools/merge_sv.cwl:8]: any plans to use the sbg repo instead of dockerhub?

**Review by mark-welsh — APPROVED** [review]
R scripts look fine. my only concern with the CWL is that not all of the files are using v1.2, but I will defer to you on that. if the tests are running fine that its okay with me

the only comment I have is to be mindful of pulling from dockerhub with this instead of the SBG repo

**Comment by migbro** [comment]
Hi @zhangb1 , did you happen to see any of our comments? I still see this on my code review board, just making sure this gets through.

File: cwl/tools/merge_seg.cwl (hunk 1 above)
Comment (zhangb1) [reply cwl/tools/merge_seg.cwl:21]: sorry I totally missed the comments. https://cavatica.sbgenomics.com/u/zhangb1/data-assembly-dev/tasks/f89058d6-08ad-4bc4-9263-ee8e181737d2/ here is the test for the WXS ...

File: cwl/tools/merge_sv.cwl (hunk 2 above)
Comment (zhangb1) [reply cwl/tools/merge_sv.cwl:8]: now will be only on dockerhub, maybe move to sbg , not sure yet

File: cwl/tools/merge_seg.cwl (hunk 1 above)
Comment (migbro) [reply cwl/tools/merge_seg.cwl:21]: Is that intentional though? The output is:
```
 new_merged_consensus_seg No value
```
in that task output when WXS is given

Comment (zhangb1) [reply cwl/tools/merge_seg.cwl:21]: Yes. the WXS mode, won't have the consensus_seg file output, it will only use the CNVkit result

**Review by migbro — APPROVED** [review]
My comments have been addressed, looks good!

**Pull request merged**